pkg/metallb/frr_test.go

```diff
@@ -0,0 +1,510 @@
+		namespace     string
+		expectedError string
+	}{
+		{
+			runTimeClient: apiClient,
+			name:          defaultFrrConfigurationName,
+			namespace:     defaultFrrConfigurationNsName,
+			expectedError: "",
+		},
+		{
+			runTimeClient: apiClient,
+			name:          "",
+			namespace:     defaultFrrConfigurationNsName,
+			expectedError: "frrConfiguration 'name' cannot be empty",
+		},
+		{
+			runTimeClient: apiClient,
+			name:          defaultFrrConfigurationName,
+			namespace:     "",
+			expectedError: "frrConfiguration 'nsname' cannot be empty",
+		},
+		{
+			runTimeClient: nil,
+			name:          defaultFrrConfigurationName,
+			namespace:     defaultFrrConfigurationNsName,
+			expectedError: "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testSettings := clients.GetTestClients(clients.TestClientParams{SchemeAttachers: frrTestSchemes})
+		testFRRConfigurationBuilder := generateFrrConfiguration(testSettings, testCase.name, testCase.namespace)
+		assert.Equal(t, testCase.expectedError, testFRRConfigurationBuilder.errorMsg)
+		assert.NotNil(t, testFRRConfigurationBuilder.Definition)
+
+		if testCase.expectedError == "" {
+			assert.Equal(t, testCase.name, testFRRConfigurationBuilder.Definition.Name)
+			assert.Equal(t, testCase.namespace, testFRRConfigurationBuilder.Definition.Namespace)
+		}
+	}
+}
+
+func TestFrrConfigurationCreate(t *testing.T) {
+	testCases := []struct {
+		testFrrConfiguration *FrrConfigurationBuilder
+		expectedError        error
+	}{
+		{
+			testFrrConfiguration: buildValidFRRConfigurationBuilder(buildFrrConfigurationTestClientWithDummyObject()),
+			expectedError:        nil,
+		},
+		{
+			testFrrConfiguration: buildInValidFRRConfigurationBuilder(buildMetalLbTestClientWithDummyObject()),
+			expectedError:        fmt.Errorf("frrConfiguration 'name' cannot be empty"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		testMetalLbBuilder, err := testCase.testFrrConfiguration.Create()
+		assert.Equal(t, err, testCase.expectedError)
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, testMetalLbBuilder.Definition.Name, testMetalLbBuilder.Object.Name)
+		}
+	}
+}
+
+func TestFrrConfigurationDelete(t *testing.T) {
+	testCases := []struct {
+		testFrrConfiguration *FrrConfigurationBuilder
+		expectedError        error
+	}{
+		{
+			testFrrConfiguration: buildValidFRRConfigurationBuilder(buildFrrConfigurationTestClientWithDummyObject()),
+			expectedError:        nil,
+		},
+		{
+			testFrrConfiguration: buildInValidFRRConfigurationBuilder(buildFrrConfigurationTestClientWithDummyObject()),
+			expectedError:        fmt.Errorf("frrConfiguration 'name' cannot be empty"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		err := testCase.testFrrConfiguration.Delete()
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Nil(t, testCase.testFrrConfiguration.Object)
+		}
+	}
+}
+
+func TestFrrConfigurationGet(t *testing.T) {
+	testCases := []struct {
```

Comment:
needs a test case for getting a nonexistent object like in [mco/mcp_test.go](https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/mco/mcp_test.go#L144-L147)

Reply:
Ok having an issue with it. Can you take a look?  Thanks